selfie.c

```diff
@@ -2308,13 +2310,35 @@ void init_system() {
   if (is_boot_level_zero()) {
     BOOTLEVELZERO = 1;
 
-    // caution: the name of the executable must not have an extension to make this work
-    // try opening executable with zeroed flags which likely fails but just on Windows
-    if (signed_less_than(sign_extend(open(selfie_name, 0, 0), SYSCALL_BITWIDTH), 0))
+    // caution: this could also be true for executables on Linux/MacOs
+    if (has_exe_file_extension(selfie_name))
       WINDOWS = 1;
   }
 }
 
+uint64_t has_exe_file_extension(char* name) {
```

Comment:
This code repeats some work done in `replace_extension`. Can the two procedures be merged?